server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -353,20 +368,52 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     /**
-     * Sets the search query for this request.
-     *
-     * @see org.elasticsearch.index.query.QueryBuilders
+     * Sets the query for this request.
      */
-    public SearchSourceBuilder query(QueryBuilder query) {
-        this.queryBuilder = query;
+    public SearchSourceBuilder query(QueryBuilder queryBuilder) {
+        subSearchSourceBuilders = new ArrayList<>();
```

Comment:
I admit this is a bit fragile, but I think this should be addressed as a follow up. The prior behavior would replace the previous query so asserting there would break updates to this method.

Reply:
I see, maybe just assert that there is a single element in the list then so it's 1:1 with the previous behaviour of replacing?